src/util/PathUtil.cpp

```diff
@@ -270,6 +270,30 @@ auto Util::ensureFolderExists(const fs::path& p) -> fs::path {
     return p;
 }
 
+auto Util::normalizeAssetPath(const fs::path& asset_path, const fs::path& base) -> std::string {
+    auto final_path = std::string{};
+
+#ifdef _WIN32
+    // On Windows, fs::relative may return an empty or malformed path if the
+    // the root_path()s are different
+    if (asset_path.root_path() == base.root_path()) {
+        final_path = fs::relative(asset_path, base).string();
+        std::replace(final_path.begin(), final_path.end(), '\\', '/');
+    }
+#else
+    // With other platforms, the root is always '/', so a relative path can
+    // always be correctly constructed
+    final_path = fs::relative(asset_path, base).string();
+#endif
+
+    // In some cases (if `asset_path` is a relative path), `final_path` may be
+    // empty, so ensure that a valid path is always returned from this function
+    if (final_path.empty())
+        final_path = fs::weakly_canonical(asset_path).string();
+
+    return final_path;
```

Comment:
[path::generic_string() and friends](https://en.cppreference.com/w/cpp/filesystem/path/generic_string) does precisely what we want here. IMO we should always use this (and its counterpart, the constructor with `format::generic_format`) when dealing with file I/O.

```suggestion
    // fs::relative() may return an empty or malformed path if the root_path()s are different
    return (asset_path.root_path() == base.root_path() ?
            fs::relative(asset_path, base) : fs::weakly_canonical(asset_path)).generic_u8string();
```